app/forms/csv_upload_form.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CSVUploadForm
+  include ActiveModel::Validations
+
+  attr_reader :file
+
+  validates_with CSVValidator
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(file)
+    @file = file
+    return false unless file && valid?
+
+    begin
+      save_keywords
+    rescue ActiveRecord::StatementInvalid
+      errors.add(:base, I18n.t('csv.validation.bad_keyword_length'))
+    end
```

Comment:
```suggestion
    save_keywords
```

I recommend handling this validation before calling `save`. In a similar way, you already have a `valid?` method and a `validator class`. You can add some checks before the saving. 

There will always be a risk of "DB error", but it is better to let the error be raised so that if you have a logging tool, you will have the full error. If we rescue the `generic` error `StatementInvalid` and assign `bad_keyword_lenght`, it might tell the user (and us) the wrong thing (maybe the error was raised due to a special char, or a corrupted CSV file, etc... be the user is informed that the keyword has the wrong length).